File: DSA/VenikaPanwar_590013755/Day_17_Question_3.java
import java.util.*;

public class Day_17_Question_3 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter a positive number:");
        int num = sc.nextInt();

        int low = 0, high = num, ans = 0;

        while (low <= high) {
            int mid = (low + high) / 2;
            if (mid * mid <= num) {
                ans = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        System.out.println("Integer square root: " + ans);
    }
}

File: DSA/khushichaturvedi_590012789/Day_17_question_3.c
#include <stdio.h>

int intSquareRoot(int n) {
    int low = 0, high = n, ans = 0;

    while (low <= high) {
        int mid = (low + high) / 2;
        long long sq = (long long)mid * mid;

        if (sq == n)
            return mid;
        else if (sq < n) {
            ans = mid;
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }

    return ans;
}

int main() {
    int num = 17;
    int result = intSquareRoot(num);

    printf("Integer square root of %d is %d\n", num, result);

    return 0;
}
